### PR TITLE
feat(Resource-status/Graph): Allow to export graph and trigger annotations

### PR DESCRIPTION
**Pull request by bdauria**
## Description

This adds a button to export graphs as PNG, and a switch to enable / disable annotations. The switch is disabled by default.

Screenshot from 2020-12-15 17-51-34

This exports the visually displayed information (annotations, as well as metrics)

## Type of change
- [x] New functionality (non-breaking change)

## Target serie
- [x] 21.04.x (master)

### CODE REVIEW

File: www/front_src/src/Resources/Details/tabs/Graph/index.tsx [hunk 1]
```diff
@@ -59,6 +83,7 @@ const defaultTimePeriod = last24hPeriod;
 const GraphTab = ({ details }: TabProps): JSX.Element => {
   const classes = useStyles();
   const { t } = useTranslation();
+  const performanceGraphRef = React.createRef<HTMLDivElement>();
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Graph/index.tsx:86]: Why not using the hook `useRef` ?

File: www/front_src/src/Resources/Details/tabs/Graph/index.tsx [hunk 2]
```diff
@@ -184,14 +216,46 @@ const GraphTab = ({ details }: TabProps): JSX.Element => {
...
+            disabled={isNil(timeline)}
+            control={
+              <Switch
+                color="primary"
+                size="small"
+                onChange={changeEventAnnotationsActive}
+              />
+            }
+            label={
+              <Typography variant="body2">{labelDisplayEvents}</Typography>
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Graph/index.tsx:230]:
```suggestion
              <Typography variant="body2">{t(labelDisplayEvents)}</Typography>
```

**Review by Thebarda — COMMENTED** [review]
Do not forget to remove the following empty file `www/front_src/src/Resources/Graph/Performance/Graph/Annotations/Area/index.tsx` ;)

**Review by Thebarda — APPROVED** [review]
LGTM
https://www.loom.com/share/d33e07ad43fc44a4aa93505d571338fa

**Review by jdelpierre — APPROVED** [review]
I would have put the switch next to the download action (all the actions at the same place)

**Pull request merged**